apis/apps/v1alpha1/cluster_types.go

```diff
@@ -278,31 +287,55 @@ type ClusterStatus struct {
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
 }
 
+// ShardingSpec defines the sharding spec.
+type ShardingSpec struct {
+	// name defines sharding template name, this name is also part of Service DNS name, so this name will comply with IANA Service Naming rule.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=22
+	// +kubebuilder:validation:Pattern:=`^[a-z]([a-z0-9\-]*[a-z0-9])?$`
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="name is immutable"
+	Name string `json:"name"`
+
+	// template defines the component template.
+	// +kubebuilder:validation:Required
+	Template ClusterComponentSpec `json:"template,omitempty"`
+
+	// shards indicates the number of component.
+	// 1. When the number of shards increases, the relevant Actions defined in the Definition will be triggered if the conditions are met. This is specifically divided into two cases:
+	//    1.1 If the Template references a ClusterDefinition through ComponentDefRef, then the postStart Action defined in the ClusterDefinition will be executed.
+	//    1.2 [Recommended] If the Template references a ComponentDefinition through ComponentDef, then the postProvision Action defined in the ComponentDefinition will be executed.
+	// 2. When the number of shards decreases, only the preTerminate Action defined in the ComponentDefinition is supported for execution if the conditions are met.
+	//    Additionally, the resources and data associated with the corresponding Component will be deleted as well.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Minimum=0
```

Comment:
set a maximum value to protect the system from some unexpected scenarios

Reply:
> maximum

set maximum to 256